### PR TITLE
export: ports formatter in invenio3

### PR DESCRIPTION
**Pull request by ioannistsanaktsidis**
Signed-off-by: Ioannis Tsanaktsidis <ioannis.tsanaktsidis@cern.ch>

### CODE REVIEW

**Comment by jmartinm** [comment]
When you port the formats from `utils` like `Bibtex` remember to port as well the tests to make sure they are working correctly.

**Comment by ioannistsanaktsidis** [comment]
:+1:

File: inspirehep/modules/formatter/config.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+FORMATTER_EXPORT_LIMIT = 5000
```
Comment (jmartinm) [inline inspirehep/modules/formatter/config.py]: We should add a docstring below every config variable to explain what they are used for. Like

```python
COLLECTIONS_CACHE_KEY = 'DynamicCollections::'
"""Key prefix added before all keys in cache server."""
```

File: inspirehep/modules/formatter/views.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+from invenio_ext.es import es
+
+from invenio_base.decorators import wash_arguments
+
+from six import text_type
+
+from invenio_base.globals import cfg
+
+import itertools
```
Comment (jmartinm) [inline inspirehep/modules/formatter/views.py]: Check this https://pypi.python.org/pypi/isort in order to sort the imports properly.

File: inspirehep/modules/formatter/views.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+from inspirehep.utils.cv_latex import Cv_latex
+from inspirehep.utils.cv_latex_html_text import Cv_latex_html_text
+
+from invenio_ext.es import es
+
+from invenio_base.decorators import wash_arguments
+
+from six import text_type
+
+from invenio_base.globals import cfg
```
Comment (jmartinm) [inline inspirehep/modules/formatter/views.py]: this is now `current_app.config`

**Comment by jacquerie** [comment]
> When you port the formats from utils like Bibtex remember to port as well the tests to make sure they are working correctly.

Uhm, maybe this can be avoided? I could resubmit #839 rebased on top of `invenio3` + @ioannistsanaktsidis's branch instead.

**Comment by ioannistsanaktsidis** [comment]
Amended the latest changes. Note that after fixing the of= parameter and the invenio_knowledge there are some parts in the code that we need to uncomment. I have placed a note there.

**Comment by ioannistsanaktsidis** [comment]
About the output formats, I noticed that some of them are really outdated, e.g hca.yml and some others. They are still  using .bft files . I think of keeping them out of the new. Suggestions ?

**Comment by ioannistsanaktsidis** [comment]
@jmartinm this one contains both my changes as well as the test from @jacquerie . Thats why we have 2 commits to be merged. Cheers.

**Comment by jmartinm** [comment]
For all these export formats, I think the best is going to be to use the `RECORDS_REST_ENDPOINTS` [configuration](https://github.com/inspirehep/inspire-next/blob/invenio3/inspirehep/config.py#L83).

There, one can define for each export type a serializer:

```
RECORDS_REST_ENDPOINTS = dict(
    literature=dict(
        pid_type='literature',
        pid_minter='inspire_recid_minter',
        pid_fetcher='inspire_recid_fetcher',
        search_index='records-hep',
        search_type='hep',
        record_serializers={
            'application/json': ('invenio_records_rest.serializers'
                                 ':json_v1_response'),
            'application/marcxml+xml': (
                'inspirehep.modules.records.serializers.marcxml_v1_response'),
            'application/x-cv+text': (
                'inspirehep.modules.records.serializers.cv_text_v1_response'),
            'application/x-bibtex': (
                'inspirehep.modules.records.serializers.bibtex_v1_response'),

        },
        search_serializers={
            'application/json': ('invenio_records_rest.serializers'
                                 ':json_v1_search'),
            'application/vnd+inspire.brief+json': ('inspirehep.modules.records.serializers'
                                                   ':json_brief_v1_search'),
        },
        list_route='/literature/',
        item_route='/literature/<pid_value>',
        default_media_type='application/json',
        max_result_window=10000,
        query_factory_imp='inspirehep.modules.search.query:inspire_query_factory',
    ),
```

The logic of the exports (bibtex and so on) can be reused.

By the way, this PR also ports the `yml` files. This was part of the `invenio-formatter` which does not exists anymore so we don't need to port them.

Shall we open a new issue for creating the export formats in the Invenio 3 way and close this one in the meantime?

**Comment by jmartinm** [comment]
There is this issue already actually https://github.com/inspirehep/inspire-next/pull/903

**Comment by ioannistsanaktsidis** [comment]
We can do that. So should I open a new one that contains only the export of the formats or I can remove the unnecessary files from this one?

**Comment by jmartinm** [comment]
The starting point would be to create a PR where you add just the serializers so that if you do a request to `/api/literature` (for a general search) or `/api/literature/123` for a given record with content type `application/x-bibtex` the bibtex is returned.

You can create a PR for each output format.

Then we will use that API from places like detailed record, angularjs search page and so on to retrieve the export formats when needed.

**Comment by ioannistsanaktsidis** [comment]
:+1:

**Comment by ioannistsanaktsidis** [comment]
I will keep this one open as a reference, because I started porting the formats in the new brief results using angular js. In the end this one will be closed and I will open a new one, adding also @jacquerie 's tests.

**Comment by jmartinm** [comment]
Superseded by https://github.com/inspirehep/inspire-next/pull/1024 , right @ioannistsanaktsidis ?

**Comment by ioannistsanaktsidis** [comment]
@jmartinm :+1:

**Comment by ioannistsanaktsidis** [comment]
We can close this one as we handiling the exports in a different PR.

**Pull request closed without merging**